### PR TITLE
Update all Cargo dependencies that require no code changes

### PR DESCRIPTION
**Pull request by fpbgg**
### Description

Updates the dependencies in all of the Cargo.toml files to the latest version that do not require a code change.

Dependencies that require a code change include:
* `mio`
* `rand`
* `rand_chacha`
* `rusqlite`
* `time`
* `base64`
* `pico-args`

### Applicable issues

#3439 

### Additional info (benefits, drawbacks, caveats)

* Keeping dependencies up-to-date ensures we're better able to respond to security releases in our dependencies

### Checklist
- [ ] Test coverage for new  or modified code paths
    * N/A
- [ ] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
    * N/A
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
    * N/A
- [ ] New integration test(s) added to `bitcoin-tests.yml`
    * N/A

### CODE REVIEW

**Comment by fpbgg** [comment]
Note: Some tests are expected to fail because I created this pull request before I could push to the main repository.

**Review by kantai — APPROVED** [review]

File: Cargo.lock [hunk 1]
```diff
@@ -85,26 +85,40 @@ checksum = "739f4a8db6605981345c5654f3a85b056ce52f37a39d34da03f25bf2151ea16e"
...
-version = "0.7.18"
+version = "0.7.20"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "1e37cfd5e7657ada45f742d6e99ca5788580b5c529dc78faf11ece6dc702656f"
+checksum = "cc936419f96fa211c1b9166887b38e5e40b19958e5b895be7c1f93adec7071ac"
 dependencies = [
  "memchr",
 ]
 
+[[package]]
```
Comment (jcnelson) [inline Cargo.lock:95]: Can you exclude this package?

File: Cargo.lock [hunk 2]
```diff
@@ -496,35 +453,74 @@ checksum = "baf1de4339761588bc0619e3cbc0120ee582ebb74b53b4efbf79117bd2da40fd"
...
+ "js-sys",
  "num-integer",
  "num-traits",
- "time 0.1.44",
+ "time 0.1.45",
+ "wasm-bindgen",
  "winapi 0.3.9",
 ]
 
+[[package]]
```
Comment (jcnelson) [inline Cargo.lock:469]: What's using `ciborium` and `ciborium-io`?

File: Cargo.lock [hunk 3]
```diff
@@ -581,21 +578,27 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
...
  "hmac",
  "percent-encoding",
- "rand 0.8.4",
+ "rand 0.8.5",
  "sha2 0.9.9",
  "time 0.2.27",
  "version_check",
 ]
 
+[[package]]
```
Comment (jcnelson) [inline Cargo.lock:591]: What's using this?

File: Cargo.lock [hunk 4]
```diff
@@ -1147,6 +1212,15 @@ dependencies = [
  "libc",
 ]
 
+[[package]]
+name = "hermit-abi"
```
Comment (jcnelson) [inline Cargo.lock:1216]: What's using this, and can we remove it?  We don't need ABI support for an obscure unikernel.

File: Cargo.lock [hunk 5]
```diff
@@ -1260,25 +1334,48 @@ dependencies = [
...
+ "android_system_properties",
+ "core-foundation-sys",
+ "iana-time-zone-haiku",
+ "js-sys",
+ "wasm-bindgen",
+ "winapi 0.3.9",
+]
+
+[[package]]
+name = "iana-time-zone-haiku"
```
Comment (jcnelson) [inline Cargo.lock:1352]: Can we remove this?  We don't support Haiku.

File: Cargo.lock [hunk 6]
```diff
@@ -1260,25 +1334,48 @@ dependencies = [
  "tokio-rustls",
 ]
 
+[[package]]
+name = "iana-time-zone"
```
Comment (jcnelson) [inline Cargo.lock:1338]: Can we remove this, or at least remove all of these needless dependencies?

File: Cargo.lock [hunk 7]
```diff
@@ -1260,25 +1334,48 @@ dependencies = [
...
+version = "0.1.53"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "64c122667b287044802d6ce17ee2ddf13207ed924c712de9a66a5814d5b64765"
+dependencies = [
```
Comment (jcnelson) [inline Cargo.lock:1342]: It's crazy that the timezone database package "needs" a javascript runtime, a WASM binding, the winapi binding, the Android binding, and the Mac OS binding.  Let's find a way to remove this.

File: Cargo.lock [hunk 8]
```diff
@@ -1418,33 +1522,61 @@ dependencies = [
  "vcpkg",
 ]
 
+[[package]]
+name = "link-cplusplus"
```
Comment (jcnelson) [inline Cargo.lock:1526]: What uses these three packages?

File: Cargo.lock [hunk 9]
```diff
@@ -2196,16 +2311,22 @@ dependencies = [
...
+checksum = "e1cf6437eb19a8f4a6cc0f7dca544973b0b78843adbfeb3683d1a94a0024a294"
 
 [[package]]
 name = "scopeguard"
 version = "1.1.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "d29ab0c6d3fc0ee92fe66e2d99f700eab17a8d57d1c1d3b748380fb20baa78cd"
 
+[[package]]
+name = "scratch"
```
Comment (jcnelson) [inline Cargo.lock:2325]: What uses this?

File: Cargo.lock [hunk 10]
```diff
@@ -2452,14 +2523,20 @@ dependencies = [
...
-checksum = "881bf8156c87b6301fc5ca6b27f11eeb2761224c7081e69b409d5a1951a70c86"
+checksum = "bdf0c33fae925bdc080598b84bc15c55e7b9a4a43b3c704da051f977469691c9"
 dependencies = [
- "digest 0.10.3",
+ "digest 0.10.6",
  "keccak",
 ]
 
+[[package]]
+name = "signature"
```
Comment (jcnelson) [inline Cargo.lock:2535]: What uses this?

**Review by jcnelson — CHANGES_REQUESTED** [review]
I didn't get a chance to go through all of them, but the Cargo.lock brings in a lot of weird packages for unknown reasons.  Can you remove them if at all possible?

**Comment by fpbgg** [comment]
Going to issue a superseding PR today to address this

**Pull request closed without merging**